modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1020,6 +1009,37 @@ private CompletableFuture<Void> scheduleTimers(CatalogZoneDescriptor zone, boole
         return allOf(futures.toArray(CompletableFuture[]::new));
     }
 
+    /**
+     * Returns metastore long view of {@link org.apache.ignite.internal.hlc.HybridTimestamp} by revision.
+     *
+     * @param revision Metastore revision.
+     * @return Appropriate metastore timestamp or -1 if revision is already compacted.
+     */
+    private long timestampByRevision(long revision) {
+        try {
+            return metaStorageManager.timestampByRevisionLocally(revision).longValue();
+        } catch (CompactedException e) {
+            if (revision > 1) {
+                LOG.warn("Unable to retrieve timestamp by revision because of meta storage compaction, [revision={}].", revision);
```

Comment:
Could you please expand log message a bit, by adding something like: "zone reset is skipped."?

Reply:
Will do in the next PR, if you don't mind